### PR TITLE
Update HTML & JS validation for work details page

### LINKED ISSUE #8826: Add more HTML input validation to work & edition edit forms
### Describe the problem that you'd like solved

There are a few opportunities to modernize the form inputs and prevent improper entries, which would be easy to implement with HTML attributes. 

### Proposal & Constraints

Specifically, we can ensure all the number entries (page number, year, weight, etc.) use the `type="number"` attribute, and all URLs use `type="url"`.

### Additional context

All the changes would be made in `web.html`, `excerpts.html` and `edition.html`, but one could certainly do similar updates to other forms where needed, either bundled with this issue or as a separate one.

Also, this would not involve adding the `type="date"` attribute to any input fields, as I assume we want to leave some flexibility re: how much the submitter knows of the date rather than requiring MM/DD/YYYY.

### PR DESCRIPTION
**Pull request by rebecca-shoptaw**
Closes #8826.

Feature. Implements URL validation on the Work Details page.

### Technical

What was initially just supposed to be adding some HTML type attributes to inputs ended up being far more complicated than expected: it turned out that the only element definitely in need of a new type was the URL input on the Work Details page, but switching to `type="url"` removed a lot of the styling and activated a jQuery plugin called `validate` we couldn't remove which generated an error message span that further disrupted the styling and spacing (and still didn't prevent a bad URL entry).

The solution was in three parts:
1. Add `type="url"` to the input in the HTML
2. Write and call a new validation function in the JavaScript to prevent bad URL submission
3. Clean up and adjust the CSS to keep styling consistent despite type change and new span

### Testing

- On the Work Details page, try to add an improperly formatted URL.
- The text should turn red and an error message should appear.
- If you try to submit with a bad URL, a second error message should appear up top, and the submission should fail.

### Screenshot

Screenshot 2024-03-05 at 2 18 21 PM

Screenshot 2024-03-05 at 2 18 29 PM

Screenshot 2024-03-05 at 2 21 24 PM

### Stakeholders

@cdrini

### CODE REVIEW

File: openlibrary/plugins/openlibrary/js/edit.js [hunk 1]
```diff
@@ -439,17 +439,33 @@ export function initEditLinks() {
         },
         validate: function(data) {
             const i18nStrings = JSON.parse(document.querySelector('#link-errors').dataset.i18n);
+            const url = data.url.trim();
+
+            function isValidURL(url) {
```
Comment (cdrini) [inline openlibrary/plugins/openlibrary/js/edit.js:444]: Toss this function at the bottom of the file; it's sufficiently general purpose, and will make this code block a little clearer if it's not here.

File: openlibrary/templates/books/edit/web.html [hunk 2]
```diff
@@ -10,13 +10,13 @@
...
-    <div id="links-form">
+    <div id="links-form" class="links-form">
         <div class="formElement">
             $:_('Please connect Open Library records to <u>good</u><span class="orange">*</span> online resources.')
         </div>
 
-        <div class="formElement">
-            <div class="formElement contentHalf">
+        <div class="formElement links-form__inputs">
+            <div class="formElemen links-form__label">
```
Comment (cdrini) [inline openlibrary/templates/books/edit/web.html:19]:
```suggestion
            <div class="formElement links-form__label">
```

File: static/css/components/form.olform.less [hunk 3]
```diff
@@ -84,16 +84,38 @@
...
+      flex: 1 1 0;
+      min-width: 250px;
+    }
+  }
+
   input[type=number],
   input[type=text],
-  input[type=password] {
+  input[type=password],
+  input.addweb {
```
Comment (cdrini) [inline static/css/components/form.olform.less:110]: I _think_ using `input[type=url]` might be ok here? We generally want all of these input fields to appear the same.

**Review by cdrini — APPROVED** [review]
Lgtm! A few small code changes; but testing all looked good!

Tested:
- ✅ Empty label + empty url + "Add Link": Shows error (label error)
- ✅ Empty label + empty url + Save page: Saving page not blocked
- ✅ Empty label + bad url + "Add Link": Shows error (for both)
- ✅ Empty label + bad url + Save page: Saving blocked
- ✅ Empty label + good url + "Add Link": Shows error (label error)
- ✅* Empty label + good url + Save page: Saving not blocked
    - I believe pre-existing issue, so not a regression
- ✅ Good label + empty url + "Add link": Shows warning (label warning)
- ✅* Good label + empty url + Save Page: Saving not blocked
    - I believe pre-existing issue.
- ✅ Good label + bad url + "Add link": Shows error (url error)
- ✅ Good label + bad url + Save page: Saving blocked
- Good label + good url + "Add link": Adds the link
- ✅* Good label + good url + Save page: Allows save
    - Note this actually doesn't save the label/url, since the user didn't click "Add link"; but that's a pre-existing issue and not a regression.

File: openlibrary/templates/books/edit/web.html (hunk 2 above)
Comment (rebecca-shoptaw) [reply openlibrary/templates/books/edit/web.html:19]: Good catch!! Thank you!

File: static/css/components/form.olform.less (hunk 3 above)
Comment (rebecca-shoptaw) [reply static/css/components/form.olform.less:110]: Very possibly! Will mess around to see if that has unfortunate effects on other pages, and switch it if it seems safe. 😊

**Review by cdrini — APPROVED** [review]
Lgtm! 🥳  Thank you @rebecca-shoptaw !

**Pull request merged**